lib/quiet_quality/message.rb

```diff
@@ -1,25 +1,49 @@
 module QuietQuality
   class Message
-    attr_accessor :annotated_line
-    attr_reader :path, :body, :start_line, :stop_line, :level, :rule
+    attr_writer :annotated_line
 
     def self.load(hash)
       new(**hash)
     end
 
     def initialize(**attrs)
       @attrs = attrs.map { |k, v| [k.to_s, v] }.to_h
-      @path = @attrs.fetch("path")
-      @body = @attrs.fetch("body")
-      @start_line = @attrs.fetch("start_line")
-      @stop_line = @attrs.fetch("stop_line", @start_line)
-      @annotated_line = @attrs.fetch("annotated_line", nil)
-      @level = @attrs.fetch("level", nil)
-      @rule = @attrs.fetch("rule", nil)
     end
 
     def to_h
       @attrs.map { |k, v| [k.to_s, v] }.to_h
     end
+
+    def path
```

Comment:
This was all because I was getting a complaint about the constructor being too complex after I added tool_name.

Reply:
Overall, I like this a lot better. But there is a functional change here - now we'll be getting errors and missing data when we try to display it, instead of when we parse it. 

Perhaps a validate method to call from the initializer, that raises if any of the required keys are not supplied, or are nil?